source/grid/tria.cc

```diff
@@ -2068,11 +2068,22 @@ namespace internal
             // need to allow that
             if (boundary_line->boundary_id != numbers::internal_face_boundary_id)
               {
-                AssertThrow (! (line->boundary_id() == numbers::internal_face_boundary_id),
-                             ExcInteriorLineCantBeBoundary(line->vertex_index(0),
-                                                           line->vertex_index(1),
-                                                           boundary_line->boundary_id));
-                line->set_boundary_id (boundary_line->boundary_id);
+                if (line->boundary_id() == numbers::internal_face_boundary_id)
+                  {
+                    if (boundary_line->manifold_id == numbers::flat_manifold_id)
+                      {
+                        AssertThrow (! (line->boundary_id() == numbers::internal_face_boundary_id),
```

Comment:
Woflgang, I recalled why the assertion is now working that way, and I would say that, to me, it's correct.
Now, if create_trangulation finds out that we want to work  on an internal line, it doesn't immediately throw the exception, but first checks if we only want to set a manifold id. In that case, it allows the operation. Otherwise, it throws the usual exception.
Just let me know if you think there's a better way to make this work.

I'll alse have a simple grid (~30 cells) I can use for a test. Just let me get familiar with the procedure to add tests and I will add that too.